src/test/java/com/microsoft/graph/serializer/DefaultSerializerTests.java

```diff
@@ -99,4 +101,24 @@ public void testDeserializeDerivedType() throws Exception {
 		assertNotNull(o);
 		assertEquals("#microsoft.graph.fileAttachment", o. get("@odata.type").getAsString());
 	}
+  
+    @Test
+    public void testSerializerCanSerializeVoid() {
+        final DefaultSerializer serializer = new DefaultSerializer(new DefaultLogger());
+        Test1 t = new Test1();
+        String json = serializer.serializeObject(t);
+        // this line will emit a warning from Java 9 about illegal access to the constructor of Void 
+        // if gson TypeAdapterFactory is not handling Void properly
+        serializer.deserializeObject(json, Test1.class);
```

Comment:
Please put asserts to make sure deserialized object is same as expected object.

Reply:
Yep ok. Serialization is not actually being tested here but rather the emission of a warning (couldn't assert this as it turned out so manual inspection of log required). I'll rename the test method to make it clearer and chuck in the extra asserts.